Dockerfile

```diff
@@ -174,6 +174,10 @@ FROM vllm-base AS test
 ADD . /vllm-workspace/
 
 # install development dependencies (for testing)
+# A newer setuptools is required for installing some test dependencies from source that do not publish python 3.12 wheels
+# This installation must complete before the test dependencies are collected and installed.
+RUN --mount=type=cache,target=/root/.cache/pip \
+    python3 -m pip install "setuptools>=74.1.1"
```

Comment:
The build without this failed with
```
[2024-09-03T22:25:50Z] #37 2.608 Collecting transformers_stream_generator (from -r requirements-test.txt (line 24))
[2024-09-03T22:25:50Z] #37 2.618   Downloading transformers-stream-generator-0.0.5.tar.gz (13 kB)
[2024-09-03T22:25:50Z] #37 2.627   Preparing metadata (setup.py): started
[2024-09-03T22:25:50Z] #37 2.646   Preparing metadata (setup.py): finished with status 'error'
[2024-09-03T22:25:50Z] #37 2.648   error: subprocess-exited-with-error
[2024-09-03T22:25:50Z] #37 2.648   
[2024-09-03T22:25:50Z] #37 2.648   × python setup.py egg_info did not run successfully.
[2024-09-03T22:25:50Z] #37 2.648   │ exit code: 1
[2024-09-03T22:25:50Z] #37 2.648   ╰─> [1 lines of output]
[2024-09-03T22:25:50Z] #37 2.648       ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
[2024-09-03T22:25:50Z] #37 2.648       [end of output]
[2024-09-03T22:25:50Z] #37 2.648   
[2024-09-03T22:25:50Z] #37 2.648   note: This error originates from a subprocess, and is likely not a problem with pip.
[2024-09-03T22:25:50Z] #37 2.657 error: metadata-generation-failed
```

I verified this was because the `vllm-base` target never updates setuptools, so it still has version `45.2.0` installed, which causes this error with a very misleading message.

Reply:
I wonder why `setuptools` is at `45.2.0` specifically and why being on an older version leads to this error message? The message doesn't mention anything about version being outdated/incompatible, only that `setuptools` is nto in the build env.

Reply:
Yeah, it took some googling around to find that others ran into the same error when setuptools is just out of date- I don't know why that's the case but I'd guess it's just some api incompatibility that's not quite correctly handled.

You can reproduce by installing setuptools==45.2.0 and then trying to install transformers-stream-generator in a fresh python 3.12 venv

> I wonder why setuptools is at 45.2.0 specifically

Yeah idk, from the build logs it looks like that's just what's installed when we do `apt-get install -y python3.12 python3.12-dev python3.12-venv`, I see this in the log: 
```
[2024-09-05T21:34:56Z] #10 11.07 Get:191 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-setuptools all 45.2.0-1ubuntu0.1 [330 kB]
```